packages/yoastseo/src/scoring/assessments/seo/TextLengthAssessment.js

```diff
@@ -52,15 +53,53 @@ export default class TextLengthAssessment extends Assessment {
 	 */
 	getResult( paper, researcher ) {
 		const wordCount = researcher.getResearch( "wordCountInText" );
-		const assessmentResult = new AssessmentResult();
+
+		if	( researcher.getConfig( "textLength" ) ) {
+			this._config = this.getLanguageSpecificConfig( researcher );
+		}
 		const calculatedResult = this.calculateResult( wordCount );
 
+		const assessmentResult = new AssessmentResult();
 		assessmentResult.setScore( calculatedResult.score );
 		assessmentResult.setText( calculatedResult.resultText );
 
 		return assessmentResult;
 	}
 
+	/**
+	 * Check if there is language-specific config, and if so, overwrite the current config with it.
+	 *
+	 * @param {Researcher} researcher The researcher to use.
+	 *
+	 * @returns {Object} The config that should be used.
+	 */
+	getLanguageSpecificConfig( researcher ) {
+		const currentConfig = this._config;
+		const languageSpecificConfig = researcher.getConfig( "textLength" );
+
+		// Don't do anything if there is no language-specific config.
+		if ( ! languageSpecificConfig ) {
+			return currentConfig;
+		}
+
+		// Check if a language has configuration for custom content types.
+		if ( languageSpecificConfig.hasOwnProperty( currentConfig.customContentType ) ) {
+			return merge( currentConfig, languageSpecificConfig[ currentConfig.customContentType ] );
+		}
+
+		// Check if a language has a default cornerstone configuration.
+		if ( currentConfig.cornerstoneContent === true && currentConfig.customContentType === "" &&
+			languageSpecificConfig.hasOwnProperty( "defaultCornerstone" ) ) {
+			return merge( currentConfig, languageSpecificConfig.defaultCornerstone );
+		}
+
+		if ( languageSpecificConfig.hasOwnProperty( "defaultAnalysis" ) ) {
+			return merge( currentConfig, languageSpecificConfig.defaultAnalysis );
+		}
+
+		return currentConfig;
```

Comment:
Same here, I think this code would never be reached, and it's also uncovered by specs.

Reply:
Fixed by returning the default language-specific config as a default 👍